### PR TITLE
Prod auth

### PR DESCRIPTION
**Pull request by Parul1999**
-- product listing
-- authentication

### CODE REVIEW

**Review by ramalingaa — APPROVED** [review]
Folder structure is good, clean code, functionalities are working.

File: src/context/products-context.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const ProductContext = createContext();
+
+const ProductProvider =({children})=>{
+    const [products,setProducts] = useState([])
+
+    useEffect( ()=>{
+        const fetchData =async ()=>{
+            try{
+                const response = await axios.get('/api/products')
+                if(response.status==200)setProducts(response.data.products)
```
Comment (rahulbakshi4) [inline src/context/products-context.js:12]: check for strict equality

File: src/pages/authentication/signup.jsx [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+   setUserData(userData=>({
+     ...userData,
+     [e.target.name]:e.target.value,
+   }))
+  }
+  
+  const handleSignUp=async ()=>{
+    try{
+       const response = await axios.post(`/api/auth/signup`,userData)
+       console.log(response)
```
Comment (rahulbakshi4) [inline src/pages/authentication/signup.jsx]: remove console.log if not required

**Review by rahulbakshi4 — APPROVED** [review]
- Filter Structure looks good
- Code is clean and easily readable
- Good use of cloudinary for image hosting

Added few suggestion 
UI looks nice 
Good to merge!!

**Pull request merged**